dask_cuda/dask_cuda_worker.py

```diff
@@ -244,160 +244,233 @@ def main(
     net_devices,
     **kwargs,
 ):
-    enable_proctitle_on_current()
-    enable_proctitle_on_children()
-
-    if tls_ca_file and tls_cert and tls_key:
-        sec = Security(
-            tls_ca_file=tls_ca_file, tls_worker_cert=tls_cert, tls_worker_key=tls_key
-        )
-    else:
-        sec = None
+    worker = CUDAWorker(
+        scheduler,
+        host,
+        nthreads,
+        name,
+        memory_limit,
+        device_memory_limit,
+        rmm_pool_size,
+        pid_file,
+        resources,
+        dashboard,
+        dashboard_address,
+        local_directory,
+        scheduler_file,
+        interface,
+        death_timeout,
+        preload,
+        dashboard_prefix,
+        tls_ca_file,
+        tls_cert,
+        tls_key,
+        enable_tcp_over_ucx,
+        enable_infiniband,
+        enable_nvlink,
+        enable_rdmacm,
+        net_devices,
+        **kwargs,
+    )
 
-    try:
-        nprocs = len(os.environ["CUDA_VISIBLE_DEVICES"].split(","))
-    except KeyError:
-        nprocs = get_n_gpus()
+    async def on_signal(signum):
+        logger.info("Exiting on signal %d", signum)
+        await worker.close()
 
-    if not nthreads:
-        nthreads = min(1, multiprocessing.cpu_count() // nprocs)
+    async def run():
+        await worker
+        await worker.finished()
 
-    memory_limit = parse_memory_limit(memory_limit, nthreads, total_cores=nprocs)
+    loop = IOLoop.current()
 
-    if pid_file:
-        with open(pid_file, "w") as f:
-            f.write(str(os.getpid()))
+    install_signal_handlers(loop, cleanup=on_signal)
 
-        def del_pid_file():
-            if os.path.exists(pid_file):
-                os.remove(pid_file)
+    try:
+        loop.run_sync(run)
+    except (KeyboardInterrupt, TimeoutError):
+        pass
+    finally:
+        logger.info("End worker")
 
-        atexit.register(del_pid_file)
 
-    services = {}
+class CUDAWorker:
+    def __init__(
+        self,
+        scheduler,
+        host,
+        nthreads,
+        name,
+        memory_limit,
+        device_memory_limit,
+        rmm_pool_size,
+        pid_file,
+        resources,
+        dashboard,
+        dashboard_address,
+        local_directory,
+        scheduler_file,
+        interface,
+        death_timeout,
+        preload,
+        dashboard_prefix,
+        tls_ca_file,
+        tls_cert,
+        tls_key,
```

Comment:
Most other classes accept a `security=` keyword that can be either a Security object or a dict of kwargs to expand into a Security object.  Thoughts on conforming to that API here?  (or changing the upstream API)

Reply:
Yes good point. I've moved these options up into the CLI function to create the security object and am now passing that into the `CUDAWorker` class via the `security` kwarg.